bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java

```diff
@@ -119,6 +127,38 @@ public static String getGdbServerCommandLine(ILaunchConfiguration configuration)
 		return lst.toArray(new String[0]);
 	}
 
+	private static boolean checkIfJtagIsAvailable()
+	{
+		EspConfigParser parser = new EspConfigParser();
+		String openOCDPath = new IDFEnvironmentVariables().getEnvValue(IDFEnvironmentVariables.OPENOCD_SCRIPTS);
+		if (!openOCDPath.isEmpty() && parser.hasBoardConfigJson())
+		{
+			return true;
+		}
+		return false;
+	}
```

Comment:
Can you extract this code and use it in both the places https://github.com/espressif/idf-eclipse-plugin/blob/b00dde41824a1df851ce0df17f4493e1ce89e9a2/bundles/com.espressif.idf.launch.serial.ui/src/com/espressif/idf/launch/serial/ui/internal/CMakeMainTab2.java#L164

Reply:
EspFlashCommandGenerator can we renamed to ESPFlashUtil and move this method here?

Reply:
@kolipakakondal good point, thanks. Fixed